packages/api/src/controllers/asset.ts

```diff
@@ -111,6 +112,48 @@ function withPlaybackUrls(ingest: string, asset: WithID<Asset>): WithID<Asset> {
   };
 }
 
+const ipfsGateway = "https://ipfs.livepeer.studio/ipfs/";
+
+type IPFSAddresses = Asset["storage"]["ipfs"]["status"]["addresses"];
+
+export function withIpfsUrls(ipfs: IPFSAddresses): IPFSAddresses {
+  if (!ipfs?.videoFileCid) {
+    return ipfs;
+  }
+  ipfs = {
+    ...ipfs,
+    videoFileUrl: `ipfs://${ipfs.videoFileCid}`,
+    videoFileGatewayUrl: pathJoin(ipfsGateway, ipfs.videoFileCid),
+  };
+  if (ipfs.nftMetadataCid) {
+    ipfs = {
+      ...ipfs,
+      nftMetadataUrl: `ipfs://${ipfs.nftMetadataCid}`,
+      nftMetadataGatewayUrl: pathJoin(ipfsGateway, ipfs.nftMetadataCid),
+    };
+  }
+  return ipfs;
+}
+
+function assetWithIpfsUrls(asset: WithID<Asset>): WithID<Asset> {
+  if (!asset?.storage?.ipfs?.status?.addresses) {
+    return asset;
+  }
+  return {
+    ...asset,
+    storage: {
+      ...asset.storage,
+      ipfs: {
+        ...asset.storage.ipfs,
+        status: {
+          ...asset.storage.ipfs.status,
+          addresses: withIpfsUrls(asset.storage.ipfs.status.addresses),
```

Comment:
there should be a better way to do this 😖

Reply:
You mean the infinite recursive spread ...?

Just realized that since I added `lodash` I can just use `_.merge` now 😌 

(or did you mean something else?)

Reply:
Just added some `lodash` sourcery here, lmk what you think!
https://github.com/livepeer/studio/pull/1178/commits/631ce6bd4bfb8f782c684d35b427dbd99889b7f9

Reply:
yeah, i meant the recursive spreading. looks better after `_.merge` :D